### PR TITLE
Clean up socket + durable IO

### PR DESCRIPTION
**Pull request by jonasbostoen**

### CODE REVIEW

**Comment by jonasbostoen** [comment]
Current dependencies on/for this PR:
* main
  * **PR #3** <a href="https://app.graphite.dev/github/pr/chainbound/msg-rs/3" target="_blank">Graphite</a> 
    * **PR #7** <a href="https://app.graphite.dev/github/pr/chainbound/msg-rs/7" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/chainbound/msg-rs/7?utm_source=stack-comment).

File: msg-socket/src/rep/driver.rs [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+                        tracing::debug!("Authentication passed for {:?} ({})", auth.id, auth.addr);
+                        this.state.stats.increment_active_clients();
+
+                        this.peer_states.insert(
+                            auth.addr,
+                            StreamNotifyClose::new(PeerState {
+                                pending_requests: JoinSet::new(),
+                                conn: Framed::new(auth.stream, reqrep::Codec::new()),
+                                addr: auth.addr,
+                                egress_queue: VecDeque::new(),
```
Comment (merklefruit) [inline msg-socket/src/rep/driver.rs:92]: prealloc would be good here as well?

Comment (jonasbostoen) [reply msg-socket/src/rep/driver.rs]: True

File: msg-transport/src/durable/session.rs [hunk 2]
```diff
@@ -17,14 +17,14 @@ pub type PendingIo<Io> = Pin<Box<dyn Future<Output = io::Result<Io>> + Send>>;
 
 /// A layer can be applied to pre-process a newly established IO object. If you need
 /// multiple layers, use a single top-level layer that contains and calls the other layers.
-pub trait Layer: 'static {
+pub trait Layer<Io: AsyncRead + AsyncWrite>: 'static {
```
Comment (merklefruit) [inline msg-transport/src/durable/session.rs:20]: Amazing!!

**Review by merklefruit — APPROVED** [review]

**Comment by jonasbostoen** [comment]
### Merge Activity

* **Oct 11, 8:24 AM**: @jonasbostoen started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/chainbound/msg-rs/7).

**Pull request merged**